react-form-controls/my-app/src/RegistrationFormControlled.tsx

```diff
@@ -0,0 +1,58 @@
+import { ChangeEvent, FormEvent, useState } from 'react';
+
+export function RegistrationFormControlled() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  const inputCn =
+    'w-full rounded-md bg-zinc-300 p-1 border-2 border-neutral-400';
+
+  function handleSubmit(e: FormEvent) {
+    e.preventDefault();
+    console.log('controlled:', username, password);
+  }
+
+  function handleUsername(e: ChangeEvent) {
```

Comment:
If you declare this is `ChangeEvent<HTMLInputElement>` you won't need to use type assertion on line 16. That is, `e.target.value` will work.

Even easier, though, is to use an arrow function in `onChange`. Then it's just
`onChange={(e) => setUsername(e.target.value)}`